src/Symfony/Component/Cache/Adapter/TagAware/FilesystemTagAwareAdapter.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Cache\Adapter\TagAware;
+
+use Symfony\Component\Cache\Adapter\TagAwareAdapterInterface;
+use Symfony\Component\Cache\Marshaller\DefaultMarshaller;
+use Symfony\Component\Cache\Marshaller\MarshallerInterface;
+use Symfony\Component\Cache\PruneableInterface;
+use Symfony\Component\Cache\Traits\FilesystemTrait;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Class FilesystemTagAwareAdapter, stores tag <> id relationship as a symlink so we don't need to fetch tags on get*().
+ *
+ * Stores tag/id information as symlinks to the cache files they refer to, in order to:
+ * - skip loading tags info on reads
+ * - be able to iterate cache to clear on-demand when invalidation by tags
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author André Rømcke <andre.romcke+symfony@gmail.com>
+ */
+class FilesystemTagAwareAdapter extends AbstractTagAwareAdapter implements TagAwareAdapterInterface, PruneableInterface
+{
+    use FilesystemTrait {
+        doSave as doSaveCache;
+        doDelete as doDeleteCache;
+    }
+
+    /**
+     * Folder used for tag symlinks.
+     */
+    private const TAG_FOLDER = 'tags';
+
+    /**
+     * @var \Symfony\Component\Filesystem\Filesystem|null
```

Comment:
FQCN not needed, use statement already available

Reply:
👍 Agreed, I'm flavoured by our own NS here.